solidity/contracts/PhasedEscrow.sol

```diff
@@ -66,8 +66,96 @@ contract CurveRewardsEscrowBeneficiary is Ownable {
         curveRewards = _curveRewards;
     }
 
-    function __escrowSentTokens(uint256 amount) external {
+    function __escrowSentTokens(uint256 amount) external onlyOwner {
         token.approve(address(curveRewards), amount);
         curveRewards.notifyRewardAmount(amount);
     }
 }
+
+/// @dev Interface of recipient contract for approveAndCall pattern.
+interface IStakerRewards {
+    function receiveApproval(
+        address _from,
+        uint256 _value,
+        address _token,
+        bytes calldata _extraData
+    ) external;
+}
+
+/// @title StakerRewardsBeneficiary
+/// @notice An abstract beneficiary contract that can receive a withdrawal phase
+///         from a PhasedEscrow contract. The received tokens are immediately 
+///         funded for a designated rewards escrow beneficiary contract.
+contract StakerRewardsBeneficiary is Ownable {
+    IERC20 public token;
+    IStakerRewards public stakerRewards;
+
+    constructor(IERC20 _token, IStakerRewards _stakerRewards) public {
+        token = _token;
+        stakerRewards = _stakerRewards;
+    }
+
+    function __escrowSentTokens(uint256 amount) external onlyOwner {
+        bool success = token.approve(address(stakerRewards), amount);
+        require(success, "Token transfer approval failed");
+        
+        stakerRewards.receiveApproval(
+            address(this),
+            amount,
+            address(token),
+            ""
+        );
+    }
+}
+
+/// @title BeaconBackportRewardsEscrowBeneficiary
+/// @notice Transfer the received tokens to a designated
+///         BeaconBackportRewardsEscrowBeneficiary contract.
+contract BeaconBackportRewardsEscrowBeneficiary is StakerRewardsBeneficiary {
+    constructor(IERC20 _token, IStakerRewards _stakerRewards)
+        public
+        StakerRewardsBeneficiary(_token, _stakerRewards)
+    {
+        token = _token;
+        stakerRewards = _stakerRewards;
```

Comment:
We don't need to set them given that we are calling the parent contract constructor, correct?

I think it should be just:
```
    constructor(IERC20 _token, IStakerRewards _stakerRewards)
        public
        StakerRewardsBeneficiary(_token, _stakerRewards) { }
```

Reply:
Correct.